### ISSUE TITLE
For mAp evaluation issue

### ISSUE THREAD

**Issue by TaoRuijie** [fuankarion/active-speakers-context#17]
Thanks for your sharing!

I search the ActivityNet official code and find there is the get_ava_active_speaker_performance.py file in https://github.com/activitynet/ActivityNet/blob/master/Evaluation/get_ava_active_speaker_performance.py, which looks like to do evaluation for ava dataset. In your code you are using sklearn metric to get the mAP results right? So may I know did you use this code to do evaluation in the process of contest?

If so, could you share the format of the result csv file? My format is looks like that:

video_id,frame_timestamp,entity_box_x1,entity_box_y1,entity_box_x2,entity_box_y2,label,entity_id,score
HV0H6oc4Kvs,960.0,0.6042350000000001,0.150289,0.701954,0.349711,NOT_SPEAKING,HV0H6oc4Kvs_0960_1020:1,0.11589885

and the ground truth csv file is "ava_activespeaker_val_augmented.csv" file, which you have shared.

Thanks for your time !

**Comment by TaoRuijie** [comment]
More detailed: I am confused about the following lines in evaluation tools:

  if df_predictions["label"].unique() != ["SPEAKING_AUDIBLE"]:
    raise ValueError(
        "Predictions CSV must contain only SPEAKING_AUDIBLE label.")

So is that mean all the label ought to be "SPEAKING_AUDIBLE"?

**Comment by fuankarion** [comment]
Hi,

You are right, we use the sklearn method,  it was just easier to plug into the train/eval code. All the publised results were obtained with the official evaluation code (the one you have in the link). Actually the sklearn metric seems a bit more strict, typically, reporting about  1% less in AP.

The prediction format is defined in 'ASC_predcition_postprocessing.py'. I'm working on another similar issue so I'll put this in the readme later today.

**Comment by TaoRuijie** [comment]
Thanks for your reply, sure. I find that the official evaluation code need to mark all the label as "SPEAKING_AUDIBLE', which is sth...confused. Yes I also tried and find the sklearn mAP is a bit higher than the official result.

I think I once did not mark all lables as "SPEAKING_AUDIBLE" in my submitted result, so the evaluation result is wrong, I change all label as "SPEAKING_AUDIBLE", now I guess the result is at least correct.

Thanks!

**Comment by fuankarion** [comment]
Oh yeah that is kind of confusing because the datasets defines 3 labels. For this project we just map one of them (Speaking but inaudible) to the silent label, after that the speaking SPEAKING_AUDIBLE make sense as it is a binary problem active speaker vs all.

**Comment by TaoRuijie** [comment]
Yes I once use the label I predict (SPEAKING_AUDIBLE  and NOT_SPEAKERING) as the label in the output csv file, then the mAP is very low (Because Recall's max value is 0.8)

Glad to figure it out~

**Comment by fuankarion** [comment]
Just to complement what I said above, you would softmax the binary output and use the value of the the active speaker prediction.
I just Added a bit of documentation on this postprocessing to the readme.

**Comment by TaoRuijie** [comment]
Yes I understand, thanks for your sharing, the related information in this topic is limited I think. Thanks to share your experience!
